### DISCUSSION TITLE
Predictions doubts!

### DISCUSSION THREAD

**Discussion by pratikgehlott** [awslabs/gluonts#1696] [Q&A]
```
I have a year of data, freq=D, prediction_length=7=context_length. 
I want to predict for the full next year. 
What would be the best way to go about this? 

From what I understand;

1) Train the model on the full year of data (with the proper params)
2) For the prediction of the first week of the new year, use the last week of the past year as input for the predictor (which was also in the train data)
3) and for the prediction of each next week, use last week's (real) data.

If so, and if giving the Predictor > context_length of data (and perhaps < context_length, haven't checked) should indeed result in significantly worse performance, then I understand.
```

Now, I have a doubt, why we are suppose to give the model past data for predictions? (According to me, the model was already trained on that, so why did supplying the data help the model?)

Also, I have a doubt, How are we suppose to make predictions on a particular itemid on a particular range, is that even possible?
For example, i have 3 itemids in my data, (1,2,3), now i need to only make future predictions on itemid 3, is that possible?

@lostella, @mbohlkeschneider any help?

Thank you

**Comment by MelchiorReihlen** [comment]
Hi, I am not sure what you want to accomplish exactly. From what I understand you want to do sth like this:

Input Data: Some value from (e.g.) 25.12.2019 - 31.12.2020 on a daily basis.
Let T_0 = 31.12.2019
predict 01.01.2020 - 07.01.2020 
using 25.12.2019 - 31.12.2019 as input.

Then,
predict 08.01.2020 - 14.01.2020
using 01.01.2020 - 07.01.2020 as input.

Is that right?

If so, `generate_rolling_dataset` (https://ts.gluon.ai/_modules/gluonts/dataset/rolling_dataset.html#generate_rolling_dataset) with the step strategy might give you the data. Afterwards you can simply call `make_evaluation_predictions` to get the predictions and ground truth.

If you only need one forecast per week, it might also be a good idea to aggregate your data on a weekly level beforehand and then forecast that :)

**Comment by MelchiorReihlen** [comment]
Regarding your second question: If the usecase is as simple as only forecasting one time series (item_id), why don't you just filter your data before handing it to the predict function?
